tests/topotests/conftest.py

```diff
@@ -15,21 +15,6 @@ def pytest_addoption(parser):
     parser.addoption('--topology-only', action='store_true',
                      help='Only set up this topology, don\'t run tests')
 
-def pytest_runtest_call():
-    """
-    This function must be run after setup_module(), it does standarized post
-    setup routines. It is only being used for the 'topology-only' option.
-    """
-    # pylint: disable=E1101
-    # Trust me, 'config' exists.
-    if pytest.config.getoption('--topology-only'):
```

Comment:
Do we want to remove this feature?

If you are going to remove the ability to run topology only, then you also need to update the docs:
https://github.com/FRRouting/frr/blob/master/tests/topotests/GUIDELINES.md#generating--obtaining-configuration-files

Note: I saw the message you sent on Slack and I think we don't need to remove anything, just change the code to avoid the pytest deprecation warning.

```
/root/persist/frr-build/tests/topotests/conftest.py:25: PytestDeprecationWarning: the `pytest.config` global is deprecated.  Please use `request.config` or `pytest_configure` (if you're a pytest plugin) instead.
```

Reply:
After a looking into pytest documentation, it seems that there is not other way to obtain command line options inside this hook. If the deprecation warning is bothering, maybe we should remove it (plus docs) until we find a better way to do it.

Reply:
So pytest is being upgraded so that this warning is being spewed all the time.  I think we should fix this problem.  I've looked at the docuemnt you referenced above and do not know what it should be rewritten to pointers on what it should be?

Reply:
The link above is for the nearest title with a link. A few paragraphs below you can find:

```
Run the topology
Topogen allows us to run the topology without running any tests, you can do that using the following example commands:

$ # Running your bootstraped topology
$ sudo pytest -s --topology-only new-topo/test_new_topo.py
$ # Running the test_template.py topology
$ sudo pytest -s --topology-only example-test/test_template.py
$ # Running the ospf_topo1.py topology
$ sudo pytest -s --topology-only ospf-topo1/test_ospf_topo1.py
Parameters explanation:

-s: actives input/output capture. This is required by mininet in order to show the interactive shell.
--topology-only: don't run any tests, just build the topology.
```

Reply:
Yeah I read that but I don't know what we want to put this as.  Do you have suggestions?

Reply:
That is/was how I developed tests for topotests... However, since nobody is using it, I think we might as well remove the entire section or rewrite it to suggest using the `tgen.mininet_cli` code call.

Answering to the question "how did you develop your topology test case?" might help to find a better wording to this section.

Note: these docs were moved to somewhere else and is now contained in the official docs:
http://docs.frrouting.org/projects/dev-guide/en/latest/topotests.html#generating-obtaining-configuration-files

(sorry I missed your question before)